File: SDKs/apideck.com/pos/8.66.1/python/src/sdk/models/shared/modifiergroup.py
from dataclasses import dataclass, field
from datetime import datetime
from marshmallow import fields
import dateutil.parser
from typing import Any,Enum,List,Optional
from dataclasses_json import dataclass_json

class ModifierGroupSelectionTypeIndicatesWhetherMultipleOptionsFromTheModifierGroupCanBeAppliedToASingleOrMultipleModifiersEnum(str, Enum):
    SINGLE = "single"
    MULTIPLE = "multiple"


@dataclass_json
@dataclass
class ModifierGroup:
    alternate_name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'alternate_name' }})
    created_at: Optional[datetime] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'created_at', 'encoder': datetime.isoformat, 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    created_by: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'created_by' }})
    deleted: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'deleted' }})
    id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'id' }})
    maximum_allowed: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'maximum_allowed' }})
    minimum_required: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'minimum_required' }})
    modifiers: Optional[List[Any]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'modifiers' }})
    name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'name' }})
    present_at_all_locations: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'present_at_all_locations' }})
    row_version: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'row_version' }})
    selection_type: Optional[ModifierGroupSelectionTypeIndicatesWhetherMultipleOptionsFromTheModifierGroupCanBeAppliedToASingleOrMultipleModifiersEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'selection_type' }})
    updated_at: Optional[datetime] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'updated_at', 'encoder': datetime.isoformat, 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    updated_by: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'updated_by' }})
    

File: SDKs/apideck.com/pos/8.42.0/go/pkg/models/shared/modifiergroup.go
package shared

import (
	"time"
)

type ModifierGroupSelectionTypeIndicatesWhetherMultipleOptionsFromTheModifierGroupCanBeAppliedToASingleOrMultipleModifiersEnum string

const (
	ModifierGroupSelectionTypeIndicatesWhetherMultipleOptionsFromTheModifierGroupCanBeAppliedToASingleOrMultipleModifiersEnumSingle   ModifierGroupSelectionTypeIndicatesWhetherMultipleOptionsFromTheModifierGroupCanBeAppliedToASingleOrMultipleModifiersEnum = "single"
	ModifierGroupSelectionTypeIndicatesWhetherMultipleOptionsFromTheModifierGroupCanBeAppliedToASingleOrMultipleModifiersEnumMultiple ModifierGroupSelectionTypeIndicatesWhetherMultipleOptionsFromTheModifierGroupCanBeAppliedToASingleOrMultipleModifiersEnum = "multiple"
)

type ModifierGroup struct {
	AlternateName         *string                                                                                                                    `json:"alternate_name"`
	CreatedAt             *time.Time                                                                                                                 `json:"created_at"`
	CreatedBy             *string                                                                                                                    `json:"created_by"`
	Deleted               *bool                                                                                                                      `json:"deleted"`
	ID                    *string                                                                                                                    `json:"id"`
	MaximumAllowed        *int64                                                                                                                     `json:"maximum_allowed"`
	MinimumRequired       *int64                                                                                                                     `json:"minimum_required"`
	Modifiers             []interface{}                                                                                                              `json:"modifiers"`
	Name                  *string                                                                                                                    `json:"name"`
	PresentAtAllLocations *bool                                                                                                                      `json:"present_at_all_locations"`
	RowVersion            *string                                                                                                                    `json:"row_version"`
	SelectionType         *ModifierGroupSelectionTypeIndicatesWhetherMultipleOptionsFromTheModifierGroupCanBeAppliedToASingleOrMultipleModifiersEnum `json:"selection_type"`
	UpdatedAt             *time.Time                                                                                                                 `json:"updated_at"`
	UpdatedBy             *string                                                                                                                    `json:"updated_by"`
}
